### DISCUSSION TITLE
Visualization Branching

### DISCUSSION THREAD

**Discussion by kahboom** [KaotoIO/kaoto-ui#150] [Ideas]
As we begin to consider more complex integrations that involve transformations, conditionals, and EIPs, we should open up the discussion about Kaoto's UI should handle branching, from the user perspective. This discussion should not be technically-oriented, only based on visualization of branching.

## Considerations & Points of Discussion
1. Deep nesting and preventing branches from "conflicting" with one another, visually.
2. Canvas limitations. Do we allow infinite canvas for accommodating very large integrations?
3. Whether or not to allow for branches to be collapsed/expanded.

### Collapsable Branches
Though right now we are focusing on more linear integrations that are relatively small, wireframes with much more nested workflows are being made.

#### Collapsed

![Integration 2: Collapsed](https://user-images.githubusercontent.com/3844502/141145562-95255da8-9252-4987-a84f-5a20743d20cb.png)

#### Expanded

![Integration 2: Expanded](https://user-images.githubusercontent.com/3844502/141145587-10b1de8a-19b1-4023-aa6f-5f680fa544f4.png)

---

### Mini Steps for Deeper Nesting

![Deeper Nesting Collapsed](https://user-images.githubusercontent.com/3844502/142191406-b97f16aa-a6e5-4450-a5ef-1d2d7a6f1967.png)

![Deeper Nesting Expanded](https://user-images.githubusercontent.com/3844502/142189308-081a3188-fd25-423a-866f-6112ab09ec6d.png)

**Comment by brunoNetId** [comment]
Integration processes may become large. Best practices recommend defining small Camel routes, but the reality is that users end up using the tool in ways you wouldn't have expected.
I think it is important to ensure your canvas is not limited in size, and allow users to move around it, also to zoom in/out.
It's important to be able to display the entire flow. Many customers use GUIs also as documentation tools, to export images so that they can include them in decks and documents to their colleagues and superiors.
